pkg/operator/ceohelpers/machine_api.go

```diff
@@ -0,0 +1,41 @@
+package ceohelpers
+
+import (
+	machinelistersv1beta1 "github.com/openshift/client-go/machine/listers/machine/v1beta1"
+
+	"k8s.io/apimachinery/pkg/labels"
+	"k8s.io/client-go/tools/cache"
+	"k8s.io/utils/pointer"
+)
+
+// IsMachineAPIFunctionalWrapper returns a function for determining if the Machine API is functional.
+// As of today Machine API is functional when we find Machines resources in the Running state.
+//
+// Note:
+//  since this function needs to take into account only master machine objects
+//  make sure machineInformer contain only filtered data
+//  otherwise it might be expensive to react to every node update in larger installations
+func IsMachineAPIFunctionalWrapper(masterMachineInformer cache.SharedInformer, masterMachineLister machinelistersv1beta1.MachineLister) func() (bool, error) {
+	return func() (bool, error) {
+		if !masterMachineInformer.HasSynced() {
+			return false, nil
+		}
+		machines, err := masterMachineLister.List(labels.Everything())
```

Comment:
Is it worth filtering on the master label here as a safety net?

Reply:
yes, we are going to use a filtered informer - https://github.com/openshift/cluster-etcd-operator/pull/737/files#diff-0d623dfd885adb20f991bda4c2453aebd732ca6dbb4d1d4be6e79805c3b48de6R81